src/components/ConfirmWithAuthModal.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { walletOpenAsyncUnsafe } from '@alephium/sdk'
+import { useEffect, useState } from 'react'
+import { Alert } from 'react-native'
+import styled from 'styled-components/native'
+
+import { errorInstructionSet, pinLength } from '../screens/LoginScreen'
+import { getStoredActiveWallet } from '../storage/wallets'
+import { mnemonicToSeed, pbkdf2 } from '../utils/crypto'
+import PinCodeInput from './inputs/PinCodeInput'
+import ModalWithBackdrop from './ModalWithBackdrop'
+import SpinnerModal from './SpinnerModal'
+import CenteredInstructions, { Instruction } from './text/CenteredInstructions'
+
+interface ConfirmWithAuthModalProps {
+  onConfirm: () => void
+  onCancel: () => void
+}
+
+const firstInstructionSet: Instruction[] = [{ text: 'Please enter your pin', type: 'primary' }]
+
+const ConfirmWithAuthModal = ({ onConfirm, onCancel }: ConfirmWithAuthModalProps) => {
+  const [pinCode, setPinCode] = useState('')
+  const [shownInstructions, setShownInstructions] = useState(firstInstructionSet)
+  const [activeWalletEncryptedMnemonic, setActiveWalletEncryptedMnemonic] = useState<string>()
+  const [loading, setLoading] = useState(false)
+
+  useEffect(() => {
+    const getWallet = async () => {
+      try {
+        const storedActiveWallet = await getStoredActiveWallet()
+
+        if (storedActiveWallet) {
+          if (storedActiveWallet.authType === 'biometrics') {
+            onConfirm()
+          } else if (storedActiveWallet.authType === 'pin') {
+            setActiveWalletEncryptedMnemonic(storedActiveWallet.mnemonic)
+          }
+        }
+      } catch (e: unknown) {
```

Comment:
Typing errors strictly is often complicated (I stumbled upon this issue many times...).
Here, I feel like we can do better, as the expo packages are all written in Typescript, including `expo-local-authentication`. 
After a quick look, I see that this package exports potentially interesting types:
```ts
export { LocalAuthenticationOptions, AuthenticationType, LocalAuthenticationResult, SecurityLevel };
```

It may be worth taking a quickly look to see if this can be improved. If too complicated, let's not bother too much and just keep it as it is right now!

Reply:
I will have a look at it! However, the error I am trying to catch here is an error from `expo-secure-store` and specifically an error from `SecureStore.getItemAsync`. `expo-local-authentication` is not used here.

The docs of `expo-secure-store` for `getItemAsync` say:

> The promise will reject if an error occurred while retrieving the value.

https://docs.expo.dev/versions/latest/sdk/securestore/

Looking at the source code, however, I can see that they do have some error codes. They are just not exported.

<img width="480" alt="image" src="https://user-images.githubusercontent.com/1579899/200773021-3ce89d25-0296-42f8-ad04-9aafc8ffc8f1.png">

Reply:
I created this issue https://github.com/alephium/mobile-wallet/issues/98 and I'll merge this PR, if that's ok :)